### PR TITLE
add cname dissection to ndpi dns

### PR DESCRIPTION
**Pull request by hl33ta**

### CODE REVIEW

File: src/lib/protocols/dns.c [hunk 1]
```diff
@@ -305,17 +305,12 @@ static int search_valid_dns(struct ndpi_detection_module_struct *ndpi_struct,
...
-		continue; /* Skip CNAME */
-	      }
-
 	      if((((rsp_type == 0x1) && (data_len == 4)) /* A */
 		  || ((rsp_type == 0x1c) && (data_len == 16)) /* AAAA */
 		  )) {
 		memcpy(&flow->protos.dns.rsp_addr, packet->payload + x, data_len);
 	      }
-          else if (rsp_type == 0x0c)
+          else if (rsp_type == 0x0c || rsp_type == 0x05) // revers
```
Comment (spendletonliveaction) [inline src/lib/protocols/dns.c]: is this comment supposed to be reverse?

Comment (hl33ta) [reply src/lib/protocols/dns.c]: fixed in recent push

File: src/lib/protocols/dns.c [hunk 2]
```diff
@@ -305,17 +305,12 @@ static int search_valid_dns(struct ndpi_detection_module_struct *ndpi_struct,
...
-		continue; /* Skip CNAME */
-	      }
-
 	      if((((rsp_type == 0x1) && (data_len == 4)) /* A */
 		  || ((rsp_type == 0x1c) && (data_len == 16)) /* AAAA */
 		  )) {
 		memcpy(&flow->protos.dns.rsp_addr, packet->payload + x, data_len);
 	      }
-          else if (rsp_type == 0x0c)
+          else if (rsp_type == 0x0c || rsp_type == 0x05) // reverse dns lookup and cname record processing
```
Comment (shla2022) [inline src/lib/protocols/dns.c]: On lines 300-353, indentation doesn't make sense with surrounding code, making it painful to follow. This problem doesn't seem to be present in the upstream 4.2 code. Please re-format this part of the function and  then confirm that indentation is consistent and readable regardless of what editor is used.

**Comment by shla2022** [comment]
Minor comment relating to style.

File: src/lib/protocols/dns.c (hunk 2 above)
Comment (hl33ta) [reply src/lib/protocols/dns.c]: kk

File: src/lib/protocols/dns.c (hunk 1 above)
Comment (shla2022) [reply src/lib/protocols/dns.c]: Confirmed

**Review by shla2022 — APPROVED** [review]
⚠️ I suggest squashing on merge to reduce multiple commits here.
Uses existing code and appears compliant with expected interface. LGTM.

**Pull request merged**